presto-spi/src/main/java/com/facebook/presto/spi/type/Varchars.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.type;
+
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+
+import static io.airlift.slice.SliceUtf8.offsetOfCodePoint;
+import static java.util.Objects.requireNonNull;
+
+public final class Varchars
```

Comment:
I created this class by analogy to `Decimals`. We decided to introduce such utility class as a replacement for static methods in type implementation classes such as `LongDecimalType` or `ShortDecimalType`, because we wanted to hide such implementation details. But i agree that we have a bit different situation with `VARCHAR`. So, if you think it is better i will move this static methods directly to `VarcharType` class.